com.unity.shadergraph/Editor/ShaderFoundry/Linker/Properties/PropertyDeclarations.cs

```diff
@@ -0,0 +1,382 @@
+using System;
+using System.Collections.Generic;
+
+namespace UnityEditor.ShaderFoundry
+{
+    class PropertyDeclarations
+    {
+        static Dictionary<string, Func<FieldPropertyContext, FieldPropertyData, bool>> TypeHandlers = new Dictionary<string, Func<FieldPropertyContext, FieldPropertyData, bool>>
+        {
+            {"bool", HandleBoolType },
+            {"int", HandleIntegerType },
+            {"float", HandleFloatType },
+            {"float2", HandleFloat2Type },
+            {"float3", HandleFloat3Type },
+            {"float4", HandleFloat4Type },
+            {"UnityTexture2D", HandleUnityTexture2D },
+            {"UnityTexture2DArray", HandleUnityTexture2DArray },
+            {"UnityTextureCube", HandleUnityTextureCube },
+            {"UnityTexture3D", HandleUnityTexture3D },
+            {"VTPropertyWithTextureType", HandleVirtualTexture },
```

Comment:
Yeah, I kind of wanted to re-address virtual textures (you'll notice the native change doesn't actually add this type) when I got back to the virtual texture task. There was a lot I had already done here though that I wanted to keep in since it was one of the few things that flexed some of the necessary changes (such as the uniform read delegate) so I left it in.

I'll re-visit the name in this task: https://jira.unity3d.com/browse/PLAT-1043